checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -260,13 +264,17 @@ private boolean canBeSatisfiedThrough(LocalVariableNode localVariableNode) {
           TypeElement typeElt = TypesUtils.getTypeElement(reference.getType());
           if (typeElt == null) {
             // typeElt is null if reference.getType() was not a class, interface, annotation type,
-            // or
-            // enum---that is, was not an annotatable type.
+            // or enum -- that is, was not an annotatable type.
+            // TODO: Why use TOP?  I think it would be better to throw an exception, if this is
+            // something that should not happen.
             // That shouldn't happen, but if it does fall back to a safe default (i.e. top).
             mcAnno = mustCallAnnotatedTypeFactory.TOP;
           } else {
+            // TODO: Do we have an answer for the below question?
             // TODO: Why does this happen sometimes?
             if (typeElt.asType().getKind() == TypeKind.VOID) {
+              // TODO: Why does this return immediately rather than using `continue` or setting
+              // `mcAnno` to BOTTOM?
```

Comment:
If the one of the Obligation's resource aliases is `VOID`, then that implies that there is no actual must-call obligation, because methods cannot be called on void types. (Recall that there is no guarantee that every Obligation will correspond to an actual resource, instead the analysis guarantees that every real resource will have one or more Obligations. So, it's possible to have spurious Obligations that don't actually need to be fulfilled, and this code can be viewed as removing some of them.)